data-prepper-plugins/prometheus-sink/README.md

```diff
@@ -0,0 +1,154 @@
+# Prometheus Sink
+
+This is the Data Prepper Prometheus sink plugin that sends records to http/https endpoints. You can use the sink to send data to arbitrary HTTP Endpoints which can be backed by prometheus.
+
+
+## Usages
+
+The Prometheus sink should be configured as part of Data Prepper pipeline yaml file.
+
+### Response status
+
+* `200`: the request data has been successfully pushed to http endpoint.
+* `500`: internal server error while process the request data.
+* `400`: bad request error
+* `404`: the http endpoint is not reachable
+* `501`: the server does not recognize the request method and is incapable of supporting it for any resource
+
+### HTTP Basic authentication
+```
+pipeline:
+  ...
+  sink:
+  - prometheus:
+      authentication:
+        http-basic:
+          username: my-user
+          password: my_s3cr3t
+```
+
+### HTTP Bearer token authentication
+```
+pipeline:
+  ...
+  sink:
+  - prometheus:
+      authentication:
+         bearer-token:
+            client_id: 0oaafr4j79grYGC5d7
+            client_secret: fFel-3FutCXAOndezEsOVlght6D6DR4OIt7G5D1_oJ6YtgU17JdyXmGf0M
+            token_url: https://localhost/oauth2/default/v1/token
+            grant_type: client_credentials
+            scope: prometheusSink
+```
+
+## Configuration
+
+- `url` The http/https endpoint url which can bee backed by prometheus.
+
+- `encoding`  Default is snappy
+
+- `content-type` Default is application/x-protobuf
+
+- `remote-write-version` : Prometheus Remote.Writer version Version, Default is 0.1.0
+
+- `proxy`(optional): A String of the address of a forward HTTP proxy. The format is like "<host-name-or-ip>:\<port\>". Examples: "example.com:8100", "http://example.com:8100", "112.112.112.112:8100". Note: port number cannot be omitted.
+
+- `http_method` (Optional) : HttpMethod to be used. Default is POST.
+
+- `auth_type` (Optional): Authentication type configuration. By default, this runs an unauthenticated server.
+
+- `username`(optional): A string of username required for basic authentication
+
+- `password`(optional): A string of password required for basic authentication
+
+- `client_id`: It is the client id is the public identifier of your authorization server.
+
+- `client_secret` : It is a secret known only to the application and the authorization server.
+
+- `token_url`: The End point URL of the OAuth server.(Eg: /oauth2/default/v1/token)
+
+- `grant_type` (Optional) : This grant type refers to the way an application gets an access token. Example: client_credentials/refresh_token
+
+- `scope` (Optional) : This scope limit an application's access to a user's account.
+
+- `aws_sigv4`: A boolean flag to sign the HTTP request with AWS credentials. Default to `false`.
```

Comment:
We should use `aws:` property for sigv4 and not use `aws_`.

Reply:
@dinujoh Thanks for your review. We are following the other plugins (opensearch, httpsink etc.) key naming conventions to define the aws_sigv4 property.

Reply:
Resolved